src/logic/Utils.ts

```diff
@@ -186,16 +186,41 @@ export const generateSchemesOptions = (schemes: SchemeIni) => {
   ));
 };
 
-// Reset any Marketplace localStorage keys (effectively resetting it completely)
-export const resetMarketplace = () => {
+/**
+ * Reset Marketplace localStorage keys
+ * @param category The category to reset. If not provided, reset all categories.
+ */
+export const resetMarketplace = (category: "extensions" | "snippets" | "theme" | "all" = "all") => {
   console.debug("Resetting Marketplace");
 
-  // Loop through and reset marketplace keys
-  Object.keys(localStorage).forEach((key) => {
-    if (key.startsWith("marketplace:")) {
-      localStorage.removeItem(key);
-      console.debug(`Removed ${key}`);
-    }
+  const keysToRemove: string[] = [];
+
+  // If have category, just reset that category
+  if (category === "extensions") {
```

Comment:
Refactoring this to `switch ... case` should make it look visually better imo

Reply:
Switch statements always look gross imo, with the extra break and weird indentation and colon syntax. They're also just longer.

Reply:
Agree to disagree tbh. That Python-ish approach looks nice to me.

It's generally better practice also since we're doing multiple case comparisons of the same data type.